errors/list.go

```diff
@@ -62,14 +62,21 @@ func (l *List) Error() string {
 	return fmt.Sprintf("%s (and %d elided errors)", errmsg, len(l.errs)-1)
 }
 
-// Errors returns all errors contained on the list that are of the type Error
-// or that have an error of type Error wrapped inside them.
-// Any other errors will be ignored.
-func (l *List) Errors() []*Error {
-	var errs []*Error
+// Errors returns all errors contained on the list.
+// It flattens out the wrapped error lists inside *error.Error.
+func (l *List) Errors() []error {
+	var errs []error
 	for _, err := range l.errs {
-		var e *Error
-		if errors.As(err, &e) {
+		switch e := err.(type) {
+		case *Error:
+			if el, ok := e.Err.(*List); ok {
```

Comment:
I'm curious with removing errors.As and using the access directly to the wrapped error instead. We want to always force that the depth must be 1 ? I'm not getting why errors.As was dropped here and in other places 🤔. It will unwrap and get the list when it is wrapped on the error itself + it will recursively search for the list deeper if necessary through unwrapping, what do we gain by not allowing further unwrapping ?

Reply:
this was a brain fart =) I did this change for cosmetic purposes.. to avoid the if-elses and handle them in a switch but forgot that `errors.As()` also unwraps the error. I reverted this, thanks for catching.